controllers/base.go

```diff
@@ -38,6 +40,9 @@ func (c *CommonController) Render() error {
 type BaseController struct {
 	beego.Controller
 	i18n.Locale
+	SelfRegistration bool
```

Comment:
Can we change "SelfRegistration" and "AuthMode" to global variables? We will read them from env only once in this way.